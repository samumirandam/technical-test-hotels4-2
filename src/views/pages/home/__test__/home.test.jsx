/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { render, screen } from '@utils/test-utils';

import Home from '../index';

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Home {...setupStore} />, {});
};

describe('Test for Home page component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByText('Hotels in:')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render hotel list without errors', () => {
    const props = {
      hotelList: {
        data: [{ name: 'hotel1' }],
      },
    };
    setup(props);
    expect(screen.getByText('Hotels in:')).toBeTruthy();
  });

  test('Should render loading state', () => {
    const props = {
      hotelList: {
        isLoading: true,
      },
    };
    setup(props);
    expect(screen.getByTestId('Loader')).toBeTruthy();
  });

  test('Should render error state', () => {
    const props = {
      hotelList: {
        isError: true,
        errorDetail: 'test error',
      },
    };
    setup(props);
    expect(screen.getByText('test error')).toBeTruthy();
  });
});
