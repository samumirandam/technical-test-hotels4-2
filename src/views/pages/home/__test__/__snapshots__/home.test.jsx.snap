// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Home page component Should renders the same component 1`] = `
<div>
  <section
    class="Home"
    data-testid="Home"
  >
    <div
      class="SearchResults"
    >
      <h1
        class="SearchResults__title"
      >
        Hotels in:
         
         
      </h1>
      <p
        class="SearchResults__date"
      >
        Check in:
         
      </p>
      <p
        class="SearchResults__date"
      >
        Check out:
         
      </p>
      <p />
    </div>
    <section
      class="HotelList"
    />
  </section>
</div>
`;
